hydra-core/app/controllers/concerns/hydra/controller/download_behavior.rb

```diff
@@ -120,12 +120,18 @@ def prepare_file_headers
 
       private
 
+      # Previous implementation looped and used response.stream.write, but we
+      # found that had a problem with delay in HTTP response headers being
+      # sent, which this implementation fixed while still streaming.
+      # https://github.com/samvera/hydra-head/pull/421
       def stream_body(iostream)
-        iostream.each do |in_buff|
-          response.stream.write in_buff
+        # see https://github.com/rails/rails/issues/18714#issuecomment-96204444
+        unless response.headers["Last-Modified"] || response.headers["ETag"]
+          Rails.logger.warn("Response may be buffered instead of streaming, best to set a Last-Modified or ETag header")
         end
-      ensure
-        response.stream.close
+
+        render nothing: true
```

Comment:
I'm not totally sure what it does, I copied it from the example at https://github.com/rails/rails/issues/18714#issuecomment-95855994

I think it tells Rails not to render with it's normal rendering stuff, we're going to set response.body manually ourselves, along with response.headers manually ourselves. We don't want `render :head`, we _are_ sending a body, we're just setting it manually to a streaming IO object with `response.body =`, instead of using on rails default rendering pipeline, in order to get streaming.